src/app/reducers/comment.reducer.spec.ts

```diff
@@ -0,0 +1,266 @@
+      id: '2',
+      body: 'comment 2',
+      markup: 'MarkUp',
+      createdAt: '00:00',
+      creatorId: '1',
+      bodyRendered: '<p>comment 2</p>',
+      parentId: null,
+      selfLink: '/'
+    };
+
+    const newCommentState: CommentUI[] = [
+      {
+        id: '2',
+        body: 'comment 2',
+        markup: 'MarkUp',
+        createdAt: '00:00',
+        creatorId: '1',
+        bodyRendered: '<p>comment 2</p>',
+        parentId: null,
+        selfLink: '/'
+      },
+      {
+        id: '1',
+        body: 'comment 1',
+        markup: 'MarkUp',
+        createdAt: '00:00',
+        creatorId: '1',
+        bodyRendered: '<p>comment 1</p>',
+        parentId: null,
+        selfLink: '/'
+      }
+    ];
+
+    const getSuccessAction = new CommentActions.GetSuccess(comments);
+    const state = CommentReducer(CommentInitialState, getSuccessAction);
+
+    const addSuccessAction = new CommentActions.AddSuccess(newComment);
+    const newState = CommentReducer(state, addSuccessAction);
+
+    expect(newState).toEqual(newCommentState);
+  });
+
+  it('AddError action should return the previous state', () => {
+    const comments: CommentUI[] = [
+      {
+        id: '1',
+        body: 'comment 1',
+        markup: 'MarkUp',
+        createdAt: '00:00',
+        creatorId: '1',
+        bodyRendered: '<p>comment 1</p>',
+        parentId: null,
+        selfLink: '/'
+      }
+    ];
+
+    const getSuccessAction = new CommentActions.GetSuccess(comments);
+    const state = CommentReducer(CommentInitialState, getSuccessAction);
+
+    const addErrorAction = new CommentActions.AddError();
+    const newState = CommentReducer(state, addErrorAction);
+
+    expect(newState).toEqual(state);
+  });
+
+  it('updateSuccess action should return updates state', () => {
+    const comments: CommentUI[] = [
+      {
+        id: '1',
+        body: 'comment 1',
+        markup: 'MarkUp',
+        createdAt: '00:00',
+        creatorId: '1',
+        bodyRendered: '<p>comment 1</p>',
+        parentId: null,
+        selfLink: '/'
+      }
+    ];
+
+    const updatedComment = {
+      id: '1',
+      body: 'comment 2',
+      markup: 'MarkUp',
+      createdAt: '00:00',
+      creatorId: '1',
+      bodyRendered: '<p>comment 2</p>',
+      parentId: null,
+      selfLink: '/'
+    };
+
+    const newState: CommentUI[] = [
+      {
+        id: '1',
+        body: 'comment 2',
+        markup: 'MarkUp',
+        createdAt: '00:00',
+        creatorId: '1',
+        bodyRendered: '<p>comment 2</p>',
+        parentId: null,
+        selfLink: '/'
+      }
+    ];
+
+    const getSuccessAction = new CommentActions.GetSuccess(comments);
+    const state = CommentReducer(CommentInitialState, getSuccessAction);
+
+    const updateSuccessAction = new CommentActions.UpdateSuccess(updatedComment);
+    const updatedState = CommentReducer(state, updateSuccessAction);
+
+    expect(updatedState).toEqual(updatedState);
```

Comment:
This should be `expect(updatedState).toEqual(newState)`

Reply:
Yes, My bad.